lib/web_ui/lib/src/engine/window.dart

```diff
@@ -214,17 +217,30 @@ class EngineWindow extends ui.Window {
     scheduleFrameCallback!();
   }
 
+  @visibleForTesting
+  JsUrlStrategy? urlStrategy = ui.debugEmulateFlutterTesterEnvironment
```

Comment:
A comment explaining what the `null` value signifies would be useful. My first instinct is to make it non-null and supply an implementation specific to the test environment.